li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -146,6 +179,8 @@ private LiKafkaFederatedConsumerImpl(LiKafkaConsumerConfig configs, MetadataServ
 
     _mdsRequestTimeoutMs = configs.getInt(LiKafkaConsumerConfig.METADATA_SERVICE_REQUEST_TIMEOUT_MS_CONFIG);
     _maxPollRecordsForFederatedConsumer = configs.getInt(LiKafkaConsumerConfig.MAX_POLL_RECORDS_CONFIG);
+    _metadataMaxAgeMs = configs.getInt(LiKafkaConsumerConfig.METADATA_MAX_AGE_CONFIG);
+    _currentSubscription = new Unsubscribed();
```

Comment:
nit - you probably want to only have a singleton static instance of Unsubscribed (also make sure its "topics not found" set is immutable)

Reply:
done